queries/unix_heartbeat.kusto

```diff
@@ -2,6 +2,14 @@
 // Display the VM's reported availability during the last 5 minutes.
 Heartbeat
 | extend affected_object = toupper(tostring(split(_ResourceId, "/") [-1]))
+| join kind= leftouter (MonitoringResources_CL 
+| where TimeGenerated > ago(13h)
+| where type_s == "microsoft.compute/virtualmachines" 
+| extend vm_MonitoringResources = toupper(tostring(split(id_s, "/") [-1])) 
+| project TimeGenerated,vm_MonitoringResources, id_s, name_s, tags_managedby_s
+| summarize arg_max(TimeGenerated,*) by vm_MonitoringResources)
+on $left.affected_object == $right.vm_MonitoringResources
```

Comment:
In this case it should be possible to join on the entire resource ID instead of just the VM name. This shouldn't happen but technically it's possible to have multiple VMs with the same name in 2 subscriptions.